signingmodules/kaleidokms/internal/keymanagersigningmodule/key_manager_signing_module.go

```diff
@@ -0,0 +1,318 @@
+	"github.com/kaleido-io/paladin/toolkit/pkg/plugintk"
+	"github.com/kaleido-io/paladin/toolkit/pkg/prototk"
+	"github.com/kaleido-io/paladin/toolkit/pkg/signpayloads"
+	"github.com/kaleido-io/paladin/toolkit/pkg/verifiers"
+)
+
+type Server interface {
+	Start() error
+	Stop()
+}
+
+type keyManagerSigningModule struct {
+	bgCtx     context.Context
+	callbacks plugintk.SigningModuleCallbacks
+
+	conf       *Config
+	name       string
+	httpClient *resty.Client
+
+	keystoreName    string
+	autoKeyCreation bool
+}
+
+type ErrorResponse struct {
+	Error string `json:"error"`
+}
+
+func NewPlugin(ctx context.Context) plugintk.PluginBase {
+	return plugintk.NewSigningModule(NewKeyManagerSigningModule)
+}
+
+func NewKeyManagerSigningModule(callbacks plugintk.SigningModuleCallbacks) plugintk.SigningModuleAPI {
+	return &keyManagerSigningModule{
+		bgCtx:     context.Background(),
+		callbacks: callbacks,
+	}
+}
+
+func (rsm *keyManagerSigningModule) ConfigureSigningModule(ctx context.Context, req *prototk.ConfigureSigningModuleRequest) (*prototk.ConfigureSigningModuleResponse, error) {
+	rsm.name = req.Name
+
+	err := json.Unmarshal([]byte(req.ConfigJson), &rsm.conf)
+	if err != nil {
+		return nil, i18n.WrapError(ctx, err, msgs.MsgInvalidSigningModuleConfig)
+	}
+
+	// Set keystoreName from config if provided
+	rsm.keystoreName = rsm.conf.KeystoreName
+
+	rsm.autoKeyCreation = rsm.conf.AutoKeyCreation
+
+	if rsm.conf.HTTPConfig != nil {
+		httpClient, err := pldresty.New(ctx, rsm.conf.HTTPConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		rsm.httpClient = httpClient
+	}
+
+	return &prototk.ConfigureSigningModuleResponse{}, nil
+}
+
+func (rsm *keyManagerSigningModule) GetURI(ctx context.Context, keyName string, keyPaths []*prototk.ResolveKeyPathSegment) (string, error) {
+	path := keyPaths
+	var keystore string
+	if rsm.keystoreName != "" {
+		keystore = rsm.keystoreName
+	} else {
+		// The first key path is used as the "keystore/signer" name to where the key will be resolved
+		if len(keyPaths) == 0 {
+			return "", i18n.NewError(ctx, msgs.MsgResolvePaladinTypeMissingPaths)
+		}
+		keystore = keyPaths[0].Name // use the first segment as the keystore name
+		path = path[1:]             // remove the first segment
+	}
+
+	var keyPath string
+	for _, segment := range path {
+		keyPath = fmt.Sprintf("%s%s/", keyPath, segment.Name)
+	}
+
+	return fmt.Sprintf("kld:///keystore/%s/key/%s%s", keystore, keyPath, keyName), nil
+}
+
+func (rsm *keyManagerSigningModule) ResolveKey(ctx context.Context, req *prototk.ResolveKeyRequest) (*prototk.ResolveKeyResponse, error) {
+	if len(req.Name) == 0 {
+		return nil, i18n.NewError(ctx, msgs.MsgResolveKeyCannotBeEmpty)
+	}
+
+	if len(req.RequiredIdentifiers) == 0 {
+		return nil, i18n.NewError(ctx, msgs.MsgResolveKeyMissingIdentifiers)
+	}
+
+	if (req.RequiredIdentifiers[0].VerifierType != verifiers.ETH_ADDRESS) || (req.RequiredIdentifiers[0].Algorithm != algorithms.ECDSA_SECP256K1) {
```

Comment:
This hard coding looked odd to me at first.

But after thinking I understood that while Kaleido supports a bunch of different algorithms/public-identifiers, there's no overlap with ones we know Paladin needs right now beyond just SECP256K1.

So this is a like a temporary mapping-of-one, that in the future we'll have to work out the more dynamic mapping as our key management evolves to support the more complex scenarios.